performance_test/src/communication_abstractions/resource_manager.cpp

```diff
@@ -85,7 +85,7 @@ eprosima::fastrtps::Participant * ResourceManager::fastrtps_participant() const
   disc_config.m_simpleEDP.use_PublicationWriterANDSubscriptionReader = true;
   disc_config.leaseDuration = eprosima::fastrtps::c_TimeInfinite;
 #endif
-  PParam.rtps.builtin.domainId = m_ec.dds_domain_id();
+  PParam.domainId = m_ec.dds_domain_id();
```

Comment:
is this change compatible with the version of Fast-RTPS used in eloquent?
This code is used for both Foxy and eloquent.

Reply:
This is a good point, I think we should try to make the change compatible with the supported releases if possible. If not, we'll have to create a separate branch for eloquent, which would be unfortunate.

Reply:
Fixed.

Reply:
Fixed

Reply:
There's no compile option being set in the CMakeLists.txt called `FASTRTPS_VERSION_MAJOR`, e.g. https://github.com/ros2/performance_test/blob/ad8a90a6333ff6008fd195c3cb36ccc90e5eb2dd/performance_test/CMakeLists.txt#L61

We need some extra logic in the `CMakelists.txt` for this to work

Reply:
> There's no compile option being set in the CMakeLists.txt called `FASTRTPS_VERSION_MAJOR`, e.g.

This becomes available whenever you include something from Fast-DDS.
It is done the same way on [rmw_fastrtps](https://github.com/ros2/rmw_fastrtps/blob/4ade4a136c29c5885a2debea8b173ba29dc6ebcd/rmw_fastrtps_shared_cpp/src/participant.cpp#L168)

Reply:
Yes, sorry. I saw the other reply in https://github.com/ros2/performance_test/pull/20#issuecomment-646842473.
I'm waiting for a reply [here](https://github.com/ros2/performance_test/issues/19#issuecomment-646852500) before merging this.